### PR TITLE
Implement ORC dataset reader

### PR DESCRIPTION
**Pull request by oliverhu**
* support double, float and string types
* add sample keras unit tests

### CODE REVIEW

File: tensorflow_io/core/kernels/orc/orc_kernels.cc [hunk 1]
```diff
@@ -0,0 +1,215 @@
+#include <ctime>
```
Comment (yongtang) [inline tensorflow_io/core/kernels/orc/orc_kernels.cc:1]: @oliverhu Can you add license header like other .cc files?

File: tensorflow_io/core/kernels/orc/orc_kernels.cc [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    }
+    if (element_start == element_stop) {
+      return Status::OK();
+    }
+    for (int i = element_start; i < element_stop; i++) {
+      if (dtypes_[column_index] == DT_STRING) {
+        value->flat<tstring>().data()[i] =
+            tensors_[column_index].flat<tstring>().data()[i];
+        // LOG(INFO) << " column_index: " << column_index << " index: " << i <<
+        // " data: " << value->flat<tstring>().data()[i];
```
Comment (yongtang) [inline tensorflow_io/core/kernels/orc/orc_kernels.cc:156]: Should we uncomment so that log will show up?

File: tensorflow_io/core/ops/orc_ops.cc [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```
Comment (yongtang) [inline tensorflow_io/core/ops/orc_ops.cc:1]: Change to 2021?

File: tests/test_orc.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```
Comment (yongtang) [inline tests/test_orc.py:1]: 2021?

File: tests/test_orc.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+import os
+import numpy as np
+
+import tensorflow as tf
+import tensorflow_io as tfio
+
+
+def test_orc_input():
+    """test_pcap_input
+  """
```
Comment (yongtang) [inline tests/test_orc.py:28]: `test_orc_input`, also, I think the `"""` in ln 28 can be aligned with the previous line

**Review by yongtang — COMMENTED** [review]
@oliverhu The PR looks good though there are some minor comments.

File: tensorflow_io/core/kernels/orc/orc_kernels.cc (hunk 1 above)
Comment (oliverhu) [reply tensorflow_io/core/kernels/orc/orc_kernels.cc:1]: sounds good. We can also add plugins to check license headers?

File: tensorflow_io/core/kernels/orc/orc_kernels.cc (hunk 2 above)
Comment (oliverhu) [reply tensorflow_io/core/kernels/orc/orc_kernels.cc:156]: this is primarily for debugging purpose, let me uncomment and use DEBUG

File: tests/test_orc.py (hunk 5 above)
Comment (oliverhu) [reply tests/test_orc.py:28]: updated.. interesting, not covered by lint

**Comment by oliverhu** [comment]
@yongtang updated. Btw, will follow up in next PRs for nested schemas.

**Review by yongtang — APPROVED** [review]
LGTM, thanks for the PR and looking forward to the follow up on the nested field. 👍

**Pull request merged**